src/js/components/AppListComponent.jsx

```diff
@@ -5,19 +5,52 @@ var React = require("react/addons");
 var Messages = require("../constants/Messages");
 var AppTypes = require("../constants/AppTypes");
 var States = require("../constants/States");
-var AppComponent = require("../components/AppComponent");
+var AppListItemComponent = require("./AppListItemComponent");
 
 var AppsStore = require("../stores/AppsStore");
 var AppsEvents = require("../events/AppsEvents");
 
+var ViewTypes = require("../constants/AppListViewTypes");
+
+function createGroup(groupId, app) {
+  return {
+    id: groupId,
+    instances: app.instances,
+    tasksRunning: app.tasksRunning,
+    totalCpus: app.totalCpus,
+    totalMem: app.totalMem,
+    isGroup: true
+  };
+}
+
+function updateGroup(group, app) {
+  group.instances += app.instances;
+  group.tasksRunning += app.tasksRunning;
+  group.totalCpus += app.totalCpus;
+  group.totalMem += app.totalMem;
+  return group;
+}
+
 var AppListComponent = React.createClass({
   displayName: "AppListComponent",
 
+  contextTypes: {
+    router: React.PropTypes.func
+  },
+
   propTypes: {
+    currentGroup: React.PropTypes.string.isRequired,
     filterLabels: React.PropTypes.array,
     filterStatus: React.PropTypes.array,
     filterText: React.PropTypes.string,
-    filterType: React.PropTypes.array
+    filterType: React.PropTypes.array,
+    viewType: React.PropTypes.oneOf(ViewTypes)
+  },
+
+  getDefaultProps: function () {
+    return {
+      viewType: ViewTypes[0]
```

Comment:
Ignore this. :)
AppTypes was refactored..

But we should refactor this here to a human read-able object too, and using the full object name, like:
```
AppListViewTypes.LIST
```